parserator_web/views.py

```diff
@@ -14,11 +14,24 @@ class AddressParse(APIView):
     renderer_classes = [JSONRenderer]
 
     def get(self, request):
-        # TODO: Flesh out this method to parse an address string using the
-        # parse() method and return the parsed components to the frontend.
-        return Response({})
+        input_string = request.GET.get('address', '')
+        if not input_string:
+            raise ParseError("No address provided")
+
+        address_components, address_type = self.parse(input_string)
+
+        if address_components is None:
+            raise ParseError("Invalid address format")
+
+        return Response({
+            'input_string': input_string,
+            'address_components': address_components,
+            'address_type': address_type
+        })
 
     def parse(self, address):
-        # TODO: Implement this method to return the parsed components of a
-        # given address using usaddress: https://github.com/datamade/usaddress
+        try:
+            address_components, address_type = usaddress.tag(address)
+        except usaddress.RepeatedLabelError:
+            address_components, address_type = None, "Invalid"
```

Comment:
we can raise the ParseError here, and it will propagate to to the get method

Reply:
Good catch! I'll make that change as well. It will make the code cleaner and more uniform.